src/main/java/io/debezium/connector/vitess/VitessConnector.java

```diff
@@ -41,10 +46,49 @@ public Class<? extends Task> taskClass() {
 
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
-        if (maxTasks > 1) {
-            throw new IllegalArgumentException("Only a single connector task may be started");
+        LOGGER.info("Calculating taskConfigs for {} tasks", maxTasks);
+        List<String> shards = null;
+        String storagePerTask = properties.get(VitessConnectorConfig.OFFSET_STORAGE_PER_TASK.name());
+        if (storagePerTask != null && storagePerTask.equalsIgnoreCase("true")) {
```

Comment:
I saw you do this check twice: once here and another in the `taskConfigs` method below.
Is it cleaner to put everything about multi-task configs in that separate method after checking OFFSET_STORAGE_PER_TASK is enabled, and keep the original return value (i.e. `Collections.singletonList(properties)`) here?

Something like:

```
public List<Map<String, String>> taskConfigs(int maxTasks) {
  if (storagePerTask != null && storagePerTask.equalsIgnoreCase("true")) {
    // Put all shard listing and config generation in taskConfigsPerTask()
    return taskConfigsPerTask(...)
  }
  if (maxTasks > 1) {
    throw new IllegalArgumentException("Only a single connector task may be started");
  }
  return Collections.singletonList(properties);
}

```

This could save a lot of duplicate code and `else` branches

Reply:
I can do that.

Reply:
Actually the reason why I structured the code this way is I have a unit test which tests VitessConnector.taskConfigs()